fdbserver/DataDistribution.actor.cpp

```diff
@@ -700,6 +702,67 @@ ACTOR Future<Void> monitorPhysicalShardStatus(Reference<PhysicalShardCollection>
 	}
 }
 
+// This actor must be a singleton
+ACTOR Future<Void> prepareDataMigration(PrepareBlobRestoreRequest req,
+                                        Reference<DDSharedContext> context,
+                                        Database cx) {
+
+	try {
+		// Register as a storage server, so that DataDistributor could start data movement after
+		std::pair<Version, Tag> verAndTag = wait(addStorageServer(cx, req.ssi));
```

Comment:
it should be fine for full cluster restore. what will happen if we call this API multiple times to restore multiple different ranges? we shouldn't need to add storage server multiple times.

Reply:
It will has extra DB write cost, but the serverList will stay the same, only the tag will change. To BM, tag is not used so it doesn't matter.